### ISSUE TITLE
@protected should allow calls from closures with the same "this"

### ISSUE THREAD

**Issue by Hixie** [dart-lang/linter#257]
e.g. if `setState` is `@protected` in a super class, then code like this should work:

```dart
 foo(() => setState(() {})));
```

...because the closure is doing a call via the same implicit "this" as the object's own methods.

**Comment by zoechi** [comment]
Could be the same as #254

**Comment by pq** [comment]
Turns out the issue is in how we test for supertype set membership (net-net the issue is we're not handling generics properly).  Fix is on the way!
